sql/core/src/test/java/test/org/apache/spark/sql/execution/sort/RecordBinaryComparatorSuite.java

```diff
@@ -261,40 +263,58 @@ public void testBinaryComparatorForNullColumns() throws Exception {
   public void testBinaryComparatorWhenSubtractionIsDivisibleByMaxIntValue() throws Exception {
     int numFields = 1;
 
+    long row1Data = 11L;
+    long row2Data = 11L + Integer.MAX_VALUE;
+
+    // BinaryComparator compares longs in big-endian byte order.
```

Comment:
Would you like to add few more words? It is a bit hard to reason about this with this only sentence.

E.g. "... In little-endian sysyem, this bytes comparison does not overflow. We swap it to make sure overflow happened."

Reply:
I don't even think that's quite true. The comparison isn't endian at all  as it is byte-by-byte. But the point here is to write bytes in a certain order for the test, for sure.

Reply:
I think the overflow only happened when comparing longs? @mundaym

Reply:
I don't think overflow was the issue per se; signed vs unsigned bytes were, for sure, in the original issue. But not so much here in this test case.